### DISCUSSION TITLE
Background picture option

### DISCUSSION THREAD

**Discussion by nicovankampen** [arendst/Tasmota#11371] [Ideas]
When using a 4MB esp with spiffs there is ample room for storing a picture (for example: the apparatus its controlling) . I know, its fairly useless but might be a nice gimmick. And probably (i am not a coder) easy to add to the HTML section of tasmota.

**Comment by Jason2866** [comment]
Tasmota does not use SPIFFS. Tasmota uses littlefs (UFILESYS)
Backround pictures are already supported.
You can try with autoexec.bat and put this test files https://github.com/arendst/Tasmota/pull/10951 in Tasmotas filesystem

**Reply by sfromis** [reply]
That PR is about images on a connected display, not about including a background image in the HTML of the web UI.

**Reply by nicovankampen** [reply]
Right. By reading its cryptic description i was not entirely convinced it was what i was looking for.

**Comment by nicovankampen** [comment]
Sorry: UFILESYS.

**Comment by sfromis** [comment]
One way of sneaking an image into the web UI is to exploit the fact that Tasmota does not sanitize what you put into fields like DeviceName, meaning that if you include HTML tags there, they will be passed to the browser, including image references hosted outside of the Tasmota device itself.

**Reply by nicovankampen** [reply]
Ha Ha. not sure if that is a feature or a bug. Did you try that? Can you pass me an example?

**Reply by sfromis** [reply]
`DeviceName <img src="https://i.pinimg.com/originals/73/c1/50/73c1508014facfea2576a7ac04cd9601.jpg"/>`

**Reply by nicovankampen** [reply]
It works! . However its only an inserted picture. Not a fullscreen background.

**Reply by sfromis** [reply]
Well, if you are a bit creative with HTML, you may be able to tell the browser about you coming with a background image. Beware that HTML in DeviceName may be very "uncomfortable" to tools using DeviceName for anything.

**Reply by sfromis** [reply]
Depending on your browser, there may also be browser extensions designed for systematic modifications to specific web pages. A well-known one is Greasemonkey for Firefox.

**Reply by nicovankampen** [reply]
I already noticed some erratic behavior in the MQTT messages. So i removed it. Nice try though.
Solving it in the browser works on a per computer basis alone.

**Reply by gemu2015** [reply]
this is implemented in scripts, due to a regression it is no longer working in the current release, will update soon

<img width="856" alt="Bildschirmfoto 2021-03-18 um 10 22 07" src="https://user-images.githubusercontent.com/11647075/111602851-116b6700-87d4-11eb-9649-9ea77de58a1a.png">

**Reply by sfromis** [reply]
Does scripting also have an implementation for a background image?

**Reply by gemu2015** [reply]
not on the main page. you may define a button to a sub page. there you may have a background image, and all kinds of elements like google charts, 
\>W  define HTML on the main page restricted to tasmota html setup
\>w subpage defines a button with name subpage  with free html format

**Reply by gemu2015** [reply]
works now also on main page

\>W
$<style>
$body { background-image: url('/ufs/Aerial.jpg');}
$</style>

<img width="559" alt="Bildschirmfoto 2021-03-21 um 08 06 30" src="https://user-images.githubusercontent.com/11647075/111896819-6dc0c780-8a1c-11eb-9815-fcc72ab2fc39.png">

**Reply by DennisXK** [reply]
> this is implemented in scripts, due to a regression it is no longer working in the current release, will update soon
> 
> <img alt="Bildschirmfoto 2021-03-18 um 10 22 07" width="856" src="https://user-images.githubusercontent.com/11647075/111602851-116b6700-87d4-11eb-9649-9ea77de58a1a.png">

Hello, I would like to insert an image via the menu like you.
Unfortunately it doesn't work. Is it because of the Tasmota version? I have 14.4.2

**Reply by sfromis** [reply]
Compared to what was talked about nearly 4 years ago, Tasmota does now sanitize strings like device name, or button text, precluding *some* "sneaky" ways of inserting images. I'd not expect this to apply to using the scripting feature.
